client/ayon_traypublisher/plugins/create/create_editorial_advanced.py

```diff
@@ -0,0 +1,1341 @@
+            "shot",
+            product_name="shotMain",
+        )
+        instance_data["otioClip"] = otio.adapters.write_to_string(otio_clip)
+        c_instance = self.create_context.creators["editorial_shot"].create(
+            instance_data
+        )
+        parenting_data.update(
+            {
+                "instance_label": label,
+                "instance_id": c_instance.data["instance_id"]
+            }
+        )
+        self.log.warning(f"Created instance: {pformat(instance_data)}")
+        return c_instance
+
+    def _set_product_data_to_instance(
+        self,
+        instance_data,
+        product_type,
+        variant=None,
+        product_name=None,
+    ):
+        """Product name maker
+
+        Args:
+            instance_data (dict): instance data
+            product_type (str): product type
+            variant (Optional[str]): product variant
+                default is "main"
+            product_name (Optional[str]): product name
+
+        Returns:
+            str: label string
+        """
+        if not variant:
+            if product_name:
+                variant = product_name.split(product_type)[-1].lower()
+            else:
+                variant = "main"
+
+        folder_path = instance_data["creator_attributes"]["folderPath"]
+
+        # product name
+        product_name = product_name or f"{product_type}{variant.capitalize()}"
+        label = f"{folder_path} {product_name}"
+
+        instance_data.update(
+            {
+                "label": label,
+                "variant": variant,
+                "productType": product_type,
+                "productName": product_name,
+            }
+        )
+
+        return label
+
+    def _get_base_instance_data(
+        self,
+        otio_clip,
+        instance_data,
+        track_start_frame,
+        folder_entity,
+    ):
+        """Factoring basic set of instance data.
+
+        Args:
+            otio_clip (otio.Clip): otio clip object
+            instance_data (dict): precreate instance data
+            track_start_frame (int): track start frame
+
+        Returns:
+            dict: instance data
+
+        """
+        parent_folder_path = folder_entity["path"]
+        parent_folder_name = parent_folder_path.rsplit("/", 1)[-1]
+
+        # get clip instance properties
+        handle_start = instance_data["handle_start"]
+        handle_end = instance_data["handle_end"]
+        timeline_offset = instance_data["timeline_offset"]
+        workfile_start_frame = instance_data["workfile_start_frame"]
+        fps = instance_data["fps"]
+        variant_name = instance_data["variant"]
+
+        # basic unique folder name
+        clip_name = os.path.splitext(otio_clip.name)[0]
+        project_entity = ayon_api.get_project(self.project_name)
+
+        shot_name, shot_metadata = self._shot_metadata_solver.generate_data(
+            clip_name,
+            {
+                "anatomy_data": {
```

Comment:
I guess this is reusing some other logic, could that key `anatomy_data` be renamed to `template_data`?

Reply:
this would be out of scope for this PR